Mathlib/RepresentationTheory/Action/Limits.lean

```diff
@@ -40,6 +40,11 @@ instance [HasFiniteLimits V] : HasFiniteLimits (Action V G) where
 instance [HasLimits V] : HasLimits (Action V G) :=
   Adjunction.has_limits_of_equivalence (Action.functorCategoryEquivalence _ _).functor
 
+/-- If `V` has limits of shape `J`, so does `Action V G`.-/
+lemma hasLimitsOfShapeOfHasLimitsOfShape {J : Type w₁} [Category.{w₂} J] [HasLimitsOfShape J V] :
```

Comment:
I think this can be made an instance (with a slightly simpler name), and similarly for colimits:
```suggestion
instance hasLimitsOfShape {J : Type w₁} [Category.{w₂} J] [HasLimitsOfShape J V] :
```

Reply:
This was an instance originally (see the first commit in this PR), but it caused a instance synthesization timeout in `Mathlib.RepresentationTheory.FdRep` at `HasKernels (FdRep k G)` (see: https://github.com/leanprover-community/mathlib4/actions/runs/7468254528/job/20323323137). Bumping the maximum number of heartbeats there to 50000 fixed it, but I was not sure whether this is desirable. Maybe this can be circumvented by adding some more auxiliary instances in the `FdRep` file to make the instance search faster?

Reply:
I have tried to investigate this. I have just pushed this https://github.com/leanprover-community/mathlib4/pull/9603/commits/8c5c763932051fa37513b99c1c939f6b53785e6d which seems to fix the problem.